### PR TITLE
feat(node): v12

### PR DESCRIPTION
**Pull request by SimonSchick**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V12.md#12.0.0
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by SimonSchick** [comment]
@Flarna any breaking changes we want to make here? 😛

**Comment by SimonSchick** [comment]
@sandersn any chance we can fast track this? It's basically copy paste with only import changes :)

**Comment by Flarna** [comment]
@SimonSchick typings are not semver versions therefore I see no reason to sync breaking changes with node releases. In the end at most everything can be breaking...

**Comment by Flarna** [comment]
I think the changes for following PRs are missing here but they can be added also via separate PRs:
https://github.com/nodejs/node/pull/19691
https://github.com/nodejs/node/pull/26611
https://github.com/nodejs/node/pull/26554
https://github.com/nodejs/node/pull/26774
https://github.com/nodejs/node/pull/26387

**Comment by SimonSchick** [comment]
I haven't seen those in the changelog, I'd be willing to amend those.

**Comment by SimonSchick** [comment]
@Flarna I skipped 
nodejs/node#26554
nodejs/node#26774
for now, I cba to do this atm.

**Comment by SimonSchick** [comment]
@sandersn @rbuckton so either types-publisher is broken or the change related to `esnext` target in my PR causes it to break hardcore, please take a look.

File: types/node/crypto.d.ts [hunk 1]
```diff
@@ -565,4 +576,28 @@ declare module "crypto" {
...
+
+    /**
+     * Calculates and returns the signature for `data` using the given private key and
+     * algorithm. If `algorithm` is `null` or `undefined`, then the algorithm is
+     * dependent upon the key type (especially Ed25519 and Ed448).
+     *
+     * If `key` is not a [`KeyObject`][], this function behaves as if `key` had been
+     * passed to [`crypto.createPrivateKey()`][].
+     */
+    function sign(algorithm: string | null | undefined, data: BinaryLike, key: KeyLike | SignPrivateKeyInput): Buffer;
```
Comment (Flarna) [inline types/node/crypto.d.ts]: I think `data` should be `Binary` instead `BinarlyLike`.

File: types/node/crypto.d.ts [hunk 2]
```diff
@@ -565,4 +576,28 @@ declare module "crypto" {
...
+
+    /**
+     * Calculates and returns the signature for `data` using the given private key and
+     * algorithm. If `algorithm` is `null` or `undefined`, then the algorithm is
+     * dependent upon the key type (especially Ed25519 and Ed448).
+     *
+     * If `key` is not a [`KeyObject`][], this function behaves as if `key` had been
+     * passed to [`crypto.creratepublicKey()`][].
+     */
+    function verify(algorithm: string | null | undefined, data: BinaryLike, key: KeyLike | VerifyKeyWithOptions, signature: Binary): Buffer;
```
Comment (Flarna) [inline types/node/crypto.d.ts]: I think `data` should be `Binary` instead `BinarlyLike`.

File: types/node/globals.d.ts [hunk 3]
```diff
@@ -250,6 +250,7 @@ interface Buffer extends Uint8Array {
     writeUIntBE(value: number, offset: number, byteLength: number): number;
     writeIntLE(value: number, offset: number, byteLength: number): number;
     writeIntBE(value: number, offset: number, byteLength: number): number;
+    readBigUInt64BE(offset?: number): BigInt;
```
Comment (Flarna) [inline types/node/globals.d.ts]: I think `BigInt` is not available here, only in the ts >=3.2 variant.

File: types/node/ts3.2/globals.d.ts [hunk 4]
```diff
@@ -6,3 +6,10 @@ declare namespace NodeJS {
         bigint(): bigint;
     }
 }
+
+interface Buffer extends Uint8Array {
+    readBigUInt64BE(offset?: number): bigint;
```
Comment (Flarna) [inline types/node/ts3.2/globals.d.ts]: `readBigInt64BE`, `readBigInt64LE`, `writeBigUInt64BE` and `writeBigUInt64LE` seem to be missing here.

**Review by Flarna — APPROVED** [review]

**Comment by SimonSchick** [comment]
@rbuckton can you help out? I didn't even touch v7 typings and it's still erroring.

**Comment by sandersn** [comment]
@rbuckton and I looked at it, and it's a result of some new typings (simplecrawler) referring to `../node/v7/index.dts` in the tsconfig instead of using `<reference types="node/v7" />` in the source. It errored on checkin, but CI didn't catch it. I'm not sure why.

Anyway, it's fixed, and I re-ran all the failed builds.

**Comment by SimonSchick** [comment]
can you merge this then or do you want me to rebase?

**Comment by sandersn** [comment]
Well, now it's failing because of Microsoft/TypeScript#31119 in nodal. I haven't had a chance to fix that one yet.

**Comment by sandersn** [comment]
OK, I cleaned up the current DT failures. After you merge from master, the build should be clean.

**Pull request merged**